docs/celery.rst

```diff
@@ -0,0 +1,119 @@
+======
+Celery
+======
+
+Official Celery documentation: http://docs.celeryproject.org/en/latest/
+
+What is it
+==========
+
+Celery is a library we use to perform tasks outside the bounds of an HTTP request.
+
+
+How to use celery
+=================
+
+All celery tasks should go into a tasks.py file or tasks module in a django app.
+This ensures that ``autodiscover_tasks`` can find the task and register it with the celery workers.
+
+These tasks should be decorated with one of the following:
+
+1. ``@task`` defines a task that is called manually (with ``task_function_name.delay`` in code)
+1. ``@periodic_task`` defines a task that is called at some interval (specified by ``crontab`` in the decorator)
+1. ``@serial_task`` defines a task that should only ever have one job running at one time
+
+
+Best practices
+==============
+
+Do not pass objects to celery.
+Instead, IDs can be passed and the celery task can retrieve the object from the database using the ID.
+This keeps message lengths short and reduces burden on RabbitMQ.
+
+Do not specify ``serializer='pickle'`` for new tasks.
+This is a deprecated message serializer and by default, we now use JSON.
+
+Queues
+======
+.. csv-table:: Queues
+    :header: Queue,I/O Bound?,Target max time-to-start,Target max time-to-start comments,Description of usage,How long does the typical task take to complete?,Best practices / Notes
+
+    send_report_throttled,,hours,"30 minutes: reports should be sent as close to schedule as possible.
```

Comment:
@dannyroberts Pulled this from the sheet you created after our meeting a few weeks ago

Its not a great table, but I think that its a good start to getting these documented in an accessible way

Reply:
Is that sheet temporary? It'd be good for there to be one source of truth (and for that source of truth to be these docs).

Reply:
The source of truth for this is a [yaml file in commcare-cloud](https://github.com/dimagi/commcare-cloud/blob/master/environments/production/app-processes.yml#L7). What do you think of adding these descriptions to that file (maybe using a new/ignored key) and add a link to that file here?

Example change to that file:
```yaml
    send_report_throttled:
      description: >
        This is used specifically for domains who are abusing Scheduled Reports and
        overwhelming the background queue. See settings.THROTTLE_SCHED_REPORTS_PATTERNS
      max_time_to_start: hours
      max_time_to_start_comment: >
        30 minutes: reports should be sent as close to schedule as possible.
        EDIT: this queue only affects mvp-* and ews-ghana
      concurrency: 2
      max_tasks_per_child: 1
```

Reply:
> Is that sheet temporary? It'd be good for there to be one source of truth (and for that source of truth to be these docs).

Yeah. I believe that that sheet is temporary and the starting point of getting our celery queues under control. It felt right to put that information here as I think that's the most complete info on what queues we have available and their intended uses. And allow Danny/Platform Reliability to iterate on this as they reduce the number of queues or change them.

@millerdev I believe we're talking about two separate things. I am documenting all the queues that HQ has. The YAML file you linked to is all the queues that run on production which are not the same. For example icds_aggregation_queue is not run on production. If we're going to document it in commcare-cloud a more complete way to do so would be [here](https://github.com/dimagi/commcare-cloud/blob/93267234fc5c065c41356de6b4c30ccfa5263c7b/src/commcare_cloud/environment/schemas/app_processes.py#L67-L95) and have it autogenerated in docs. But I'm not convinced that that's the correct place for it, as I think its useful to have this as an HQ level document instead of an infrastructure level doc